### PR TITLE
SRVCOM-1451: Add serverless docs for OSD

### PR DESCRIPTION
**Pull request by abrennan89**
Jiras:
- https://issues.redhat.com/browse/SRVCOM-1451

Applies for OCP 4.9, 4.10

Direct preview links:
- All serverless content needs review from release notes down to Administer section
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/serverless-release-notes.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/discover/about-serverless.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/discover/knative-event-sources.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/discover/serverless-channels.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/develop/serverless-event-sinks.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/develop/serverless-event-delivery.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/develop/serverless-apiserversource.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/develop/serverless-pingsource.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/develop/serverless-custom-event-sources.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/develop/serverless-creating-channels.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/develop/serverless-subs.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/develop/serverless-using-brokers.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/develop/serverless-triggers.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/develop/serverless-kafka-developer.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/admin_guide/serverless-configuring-eventing-defaults.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/admin_guide/serverless-kafka-admin.html
- https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/cli_tools/advanced-kn-config.html

### CODE REVIEW

File: serverless/serverless-release-notes.adoc [hunk 1]
```diff
@@ -22,10 +22,18 @@ include::modules/serverless-api-versions.adoc[leveloffset=+1]
 
 // Deprecated and removed feature trackers
 // OCP
+ifdef::openshift-enterprise[]
 include::modules/serverless-deprecated-removed-features.adoc[leveloffset=+1]
```
Comment (abrennan89) [inline serverless/serverless-release-notes.adoc:26]: Add a new module for this, beginning at 1.21.0

**Comment by abrennan89** [comment]
Note: need to comment out content once approved before merging; OSD docs will go live for Serverless 1.21.0 in March 2022

File: serverless/admin_guide/serverless-configuring-eventing-defaults.adoc [hunk 2]
```diff
@@ -6,6 +6,13 @@ include::modules/serverless-document-attributes.adoc[]
 
 toc::[]
 
+ifdef::openshift-enterprise[]
```
Comment (mjpytlak) [inline serverless/admin_guide/serverless-configuring-eventing-defaults.adoc:9]: While you are working in the file, please add a `:_content-type:` attribute.

**Comment by abrennan89** [comment]
Currently getting errors for the preview links, will check back and update tomorrow
https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/welcome/index.html

**Comment by jrangelramos** [comment]
I miss a few sections such as functions, cli tools, security.. any plans to add that in future?
Also all lang related and next steps links on About Serverless Functions page https://deploy-preview-42074--osdocs.netlify.app/openshift-dedicated/latest/serverless/discover/serverless-functions-about.html are broken..

**Comment by abrennan89** [comment]
>I miss a few sections such as functions, cli tools, security.. any plans to add that in future?

As discussed in Slack, this is just the first PR. All of our docs that currently exist and apply to OSD will be moved eventually, but this would be too much for one PR since folks need to review it etc. These will all be added in subsequent PRs.

I didn't actually mean to include any functions stuff for now because I am trying to move things in batches for exactly that reason (to keep the links in tact). Moving one page generally means moving others that are linked to it, so I will remove the functions bit for now and add it in another PR with the rest of the functions content.

**Comment by jrangelramos** [comment]
LGTM 
QE Approved

**Comment by abrennan89** [comment]
@mjpytlak ready for another look when you have a chance, thanks

**Comment by mjpytlak** [comment]
@abrennan89 Your `indefs` look good. Peer review LGTM.

**Pull request merged**